### PR TITLE
unify copy and interrupt keybindings

### LINKED ISSUE #403: Swap copy/interrupt keybindings on windows/linux
Intercept Ctrl-C & Ctrl-Shift-C via the custom keyhandler on the xterm.js level and switch them around. Then:

| keybinding   | editor    | Juno REPL | standalone REPL |
|--------------|-----------|-----------|-----------------|
| Ctrl-C       | copy      | copy      | interrupt       |
| Ctrl-Shift-C | interrupt | interrupt | copy (windows)  |
| Ctrl-V       | paste     | paste     | paste (windows) |

### PR DESCRIPTION
**Pull request by pfitzseb**
fixes #403.

This has the added benefit that you can use one keybinding everywhere and it will do the smart thing and interrupt either in-editor or in the repl, depending on which is currently running something.

@MikeInnes: I'm not sure if this works for macs, so would be great if you could check.

### CODE REVIEW

File: lib/runtime/console2.js [hunk 1]
```diff
@@ -53,7 +53,28 @@ export function activate (ink) {
...
+        }
+        return e
+      })
+    } else if (
+      terminal.attachCustomKeyEventHandler((e) => {
+        if (e.ctrlKey && e.keyCode == 67) { // ctrl-c
+          return false
+        }
+        return e
+      })
```
Comment (MikeInnes) [inline lib/runtime/console2.js]: Did you mean to use `else if` here?

File: lib/connection/process/basic2.js [hunk 2]
```diff
@@ -57,6 +57,7 @@ function getUnix (path, args, env) {
             ty: ty,
             kill: () => ty.kill(),
             interrupt: () => ty.kill('SIGINT'),
+            interruptREPL: () => ty.write('\x03'),
```
Comment (MikeInnes) [inline lib/connection/process/basic2.js]: Why do we need this?

File: lib/runtime/console2.js (hunk 1 above)
Comment (pfitzseb) [reply lib/runtime/console2.js]: no :)

File: lib/connection/process/basic2.js (hunk 2 above)
Comment (pfitzseb) [reply lib/connection/process/basic2.js]: We don't really except for the case of running on windows without the powershell wrapper.

**Comment by pfitzseb** [comment]
We can't do this with keybindings alone because the xterm.js will process all events before bubbling them upwards for the Atom keyhandlers. ^C (and ^V) is definitely processed by the terminal app, so we need to catch those *before* they reach the terminal.

File: lib/connection/process/basic2.js (hunk 2 above)
Comment (MikeInnes) [reply lib/connection/process/basic2.js]: Ok cool. I think a quick note here would be useful so no-one removes it by accident

**Comment by MikeInnes** [comment]
Oh, one other thing -- how are we handling non-julia terminals? Presumably we should be consistent with them to, it looks like this currently only applies to the julia one

**Comment by pfitzseb** [comment]
Yeah, this is only for the Julia terminal.
And I do think it's better this way -- you basically have one global keybinding for interrupting Julia (Ctrl-Shift-C), but every other terminal isn't affected by that at all.
We could have an option for disabling the special handling of `Ctrl-C` and `Ctrl-V` in the Julia terminal I suppose.

**Comment by ChrisRackauckas** [comment]
Is this also going to catch the fact that console clearing and process killing doesn't work when the console is highlighted?

**Comment by pfitzseb** [comment]
No. Would probably be good to override those though.

**Comment by pfitzseb** [comment]
Alright, this handles ^J and ^K as well, so you can clear the console and kill julia with the normal keybinding while the repl has focus.

**Pull request merged**